src/asciidoctor-extensions-api.js

```diff
@@ -96,86 +104,86 @@ var Registry = Extensions.Registry;
 /**
  * @memberof Extensions/Registry
  */
-Registry.$$proto.block = function (name, block) {
-  if (typeof name === 'function' && typeof block === 'undefined') {
-    return Opal.send(this, 'block', null, toBlock(name));
-  } else {
-    return Opal.send(this, 'block', [name], toBlock(block));
+Registry.$$proto.block = function (name, processor) {
+  if (arguments.length === 1) {
```

Comment:
I'm wondering if we could do it in the `registerExtension` ?

Reply:
I thought about that, but it isn't always needed. The purpose of the first iteration of this function was to allow all the logic to be tested. I think we should make improvements to it in a subsequent issue.